Bottom: 8abb77054cc6a74ee11812cd4f2b16f6bcc779a2
Top:    e6ba19781ab42a7566eb15cac09be5e1c5f5ba0e
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-11 15:21:05 +0000

Revert "Remove 511-character limit on configuration file lines"

This reverts commit d3830fdd06e18e05ef986210a3a9da2f49498195.

Tested-by: Kai-Martin Knaak <knaak@iqo.uni-hannover.de>


---

diff --git a/src/hid/gtk/gui-config.c b/src/hid/gtk/gui-config.c
index eb8fbd1..f1e6f49 100644
--- a/src/hid/gtk/gui-config.c
+++ b/src/hid/gtk/gui-config.c
@@ -474,14 +474,13 @@ static void
 config_file_read (void)
 {
   FILE *f;
-  char *buf = NULL;
-  size_t n = 0;
-  gchar *option, *arg;
+  gchar buf[512], *option, *arg;
 
   if ((f = config_file_open ("r")) == NULL)
     return;
 
-  while (getline (&buf, &n, f) != -1)
+  buf[0] = '\0';
+  while (fgets (buf, sizeof (buf), f))
     {
       if (parse_option_line (buf, &option, &arg) > 0)
 	set_config_attribute (option, arg);
@@ -489,7 +488,6 @@ config_file_read (void)
       g_free (arg);
     }
 
-  free (buf);
   fclose (f);
 }
